internal/ostree/ostree.go

```diff
@@ -65,47 +72,48 @@ func ResolveRef(location, ref string) (string, error) {
 	return parent, nil
 }
 
-// ResolveParams resolves all necessary missing parameters in the given struct:
-// it sets the defaultRef if none is provided and resolves the parent commit if
+// ResolveParams resolves the ostree request parameters into a commit
+// (checksum) to be fetched and the necessary ref for the image build pipeline.
+// It uses the defaultRef if none is provided and resolves the parent commit if
 // a URL and Ref are provided. If there is an error, it will be of type
 // InvalidParameterError or ResolveRefError (from the ResolveRef function)
-func ResolveParams(params RequestParams, defaultRef string) (RequestParams, error) {
-	resolved := RequestParams{}
-	resolved.Ref = params.Ref
+func ResolveParams(params RequestParams, defaultRef string) (string, string, error) {
+	ref := params.Ref
+
 	// if ref is not provided, use distro default
-	if resolved.Ref == "" {
-		resolved.Ref = defaultRef
+	if ref == "" {
+		ref = defaultRef
 	} else if !VerifyRef(params.Ref) { // only verify if specified in params
-		return resolved, NewRefError("Invalid ostree ref %q", params.Ref)
+		return "", "", NewRefError("Invalid ostree ref %q", params.Ref)
 	}
 
 	if params.Parent != "" {
 		// parent must also be a valid ref
 		if !VerifyRef(params.Parent) {
-			return resolved, NewRefError("Invalid ostree parent ref %q", params.Parent)
+			return "", "", NewRefError("Invalid ostree parent ref %q", params.Parent)
 		}
 		if params.URL == "" {
 			// specifying parent ref also requires URL
-			return resolved, NewParameterComboError("ostree parent ref specified, but no URL to retrieve it")
+			return "", "", NewParameterComboError("ostree parent ref specified, but no URL to retrieve it")
 		}
 	}
 
-	resolved.URL = params.URL
-	if resolved.URL != "" {
-		// if a URL is specified, we need to fetch the commit at the URL
+	var checksum string
+	if params.URL != "" {
+		// if a URL is specified, we need to fetch the commit at the URL.
 		// the reference to resolve is the parent commit which is defined by
-		// the 'parent' argument
-		// if the parent argument is not specified, we use the specified ref
-		// if neither is specified, we use the default ref
-		parentRef := params.Parent
-		if parentRef == "" {
-			parentRef = resolved.Ref
+		// the 'parent' argument if present.
+		// if the parent argument is not specified, we use the specified ref.
+		// if neither is specified, we use the default ref.
+		fetchRef := params.Parent
```

Comment:
Nitpick: You added dot at the end of the some lines, so the first letter of each following sentence should be capital.